cupyx/cutensor.pyx

```diff
@@ -106,8 +46,228 @@ def check_availability(name):
     return True
 
 
-cdef class Mode:
+cdef class Handle:
+    cdef intptr_t _ptr
+
+    def __init__(self):
+        self._ptr = cutensor.create()
+
+    def __dealloc__(self):
+        cutensor.destroy(self._ptr)
+        self._ptr = <intptr_t>NULL
+
+    @property
+    def ptr(self):
+        return self._ptr
+
+
+cdef class TensorDescriptor:
+    cdef intptr_t _ptr
+    cdef int _cutensor_dtype
+
+    def __init__(self, intptr_t handle, uint32_t num_modes, intptr_t extent,
+                 intptr_t stride, int cutensor_dtype,
+                 uint32_t alignment_req=256):
+        self._ptr = cutensor.createTensorDescriptor(
+            handle, num_modes, extent, stride, cutensor_dtype, alignment_req)
+        self._cutensor_dtype = cutensor_dtype
+
+    def __dealloc__(self):
+        cutensor.destroyTensorDescriptor(self._ptr)
+        self._ptr = <intptr_t>NULL
+
+    @property
+    def ptr(self):
+        return self._ptr
+
+    @property
+    def cutensor_dtype(self):
+        return self._cutensor_dtype
+
+
+cdef class OperationDescriptor:
+    cdef intptr_t _ptr
```

Comment:
This is the correct wrapper design 👍 One feature of cuTENSOR 2 is that all operations are unified under the op desc.

One nit (no real use case yet) is to add an introspection string for self-identification ("what kind of operation am I encapsulating?").

Reply:
Thanks, you're right, it would be better to write down which opaque structure of cuTENSOR each class encapsulates, not just this class.